src/payment/types.rs

```diff
@@ -474,3 +477,475 @@ impl StrictDecode for ShortChannelId {
 impl lightning_encoding::Strategy for ShortChannelId {
     type Strategy = lightning_encoding::strategies::AsStrict;
 }
+
+#[derive(Clone, Debug, From, PartialEq, Eq)]
+pub enum AnnouncedNodeAddr {
+    /// An IPv4 address/port on which the peer is listening.
+    IPv4 {
+        /// The 4-byte IPv4 address
+        addr: [u8; 4],
+        /// The port on which the node is listening
+        port: u16,
+    },
+    /// An IPv6 address/port on which the peer is listening.
+    IPv6 {
+        /// The 16-byte IPv6 address
+        addr: [u8; 16],
+        /// The port on which the node is listening
+        port: u16,
+    },
+    /// An old-style Tor onion address/port on which the peer is listening.
+    OnionV2 {
+        /// The bytes (usually encoded in base32 with ".onion" appended)
+        addr: [u8; 10],
+        /// The port on which the node is listening
+        port: u16,
+    },
+    /// A new-style Tor onion address/port on which the peer is listening.
+    /// To create the human-readable "hostname", concatenate ed25519_pubkey,
+    /// checksum, and version, wrap as base32 and append ".onion".
+    OnionV3 {
+        /// The ed25519 long-term public key of the peer
+        ed25519_pubkey: [u8; 32],
+        /// The checksum of the pubkey and version, as included in the onion
+        /// address
+        // Optional values taken here to be compatible with Uniform encoding
+        checksum: Option<u16>,
+        /// The version byte, as defined by the Tor Onion v3 spec.
+        version: Option<u8>,
+        /// The port on which the node is listening
+        port: u16,
+    },
+}
+
+impl AnnouncedNodeAddr {
+    fn get_id(&self) -> u8 {
```

Comment:
`get_` names are not recommended in rust. I propose to have it as `into_u8`, as we do everywhere else for similar enum types.

Reply:
Done